src/back/glsl/mod.rs

```diff
@@ -2203,6 +2137,80 @@ impl<'a, W: Write> Writer<'a, W> {
         Ok(())
     }
 
+    /// Helper method used to write constant expressions
+    ///
+    /// # Notes
+    /// Adds no newlines or leading/trailing whitespace
+    fn write_const_expr(&mut self, expr: Handle<crate::Expression>) -> BackendResult {
+        self.write_possibly_const_expr(expr, &self.module.const_expressions, |writer, expr| {
+            writer.write_const_expr(expr)
+        })
+    }
+
+    /// Helper method used to write possibly constant expressions
+    ///
+    /// # Notes
+    /// Adds no newlines or leading/trailing whitespace
+    fn write_possibly_const_expr<E>(
+        &mut self,
+        expr: Handle<crate::Expression>,
+        expressions: &crate::Arena<crate::Expression>,
+        write_expression: E,
+    ) -> BackendResult
```

Comment:
I think it would work but creating and passing a new `FunctionCtx`/`ExpressionContext` doesn't feel right conceptually. They carry a bunch of other data that we don't have at the module/global level.

Reply:
I see - we need to be able to write constant expressions when we're outside of any function/entry point, so the `ty` and `info` fields of `FunctionCtx` don't have any meaningful value. It really is a *function* context.